test/test-utils.js

```diff
@@ -86,3 +90,153 @@ export function testInitializeLayer({gl, layer, viewport}) {
   }
   return failures;
 }
+
+export function testUpdateLayer({gl, layer, viewport, newProps}) {
+  gl = gl || getGLContext();
+  viewport = viewport || getViewport();
+
+  const oldContext = layer.context || {gl, viewport};
+  const context = {gl, viewport};
+  const oldProps = layer.oldProps || layer.props || {};
+
+  const mergedDefaultProps = mergeDefaultProps(layer);
+  layer.props = Object.assign({}, mergedDefaultProps, newProps);
+
+  let failures = 0;
+  try {
+    layer.updateLayer({
+      oldProps,
+      props: layer.props,
+      oldContext,
+      context,
+      changeFlags: layer.diffProps(oldProps, layer.props, context)
+    });
+  } catch (error) {
+    failures++;
```

Comment:
I guess this follows the existing code, but how could failures be more than 0 or 1? Should we use a `Boolean`?